src/analysis/validation.jl

```diff
@@ -2,102 +2,238 @@
-
-
-                if haskey(wntr_flow, link_name)
-                    link_flow_wntr[link_name][tx] = wntr_flow[link_name].values[tx]
-                    link_flow_wm[link_name][tx] = link["q"]
-
-                    link_dh_wntr[link_name][tx] = abs(wntr_dh[link_name].values[tx])
-                    link_dh_wm[link_name][tx] = link["dhn"]+link["dhp"]
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    link_name = wm_data["nw"]["1"][link_type][link_id]["source_id"][2]
+    flow_wntr = Array{Float64,1}(undef,duration)
+    flow_watermodels = Array{Float64,1}(undef,duration)
+    head_loss_wntr = Array{Float64,1}(undef,duration)
+    head_loss_watermodels = Array{Float64,1}(undef,duration)
+
+    for t in 1:duration
+        flow_wntr[t] = wntr_simulation.link["flowrate"][link_name].values[t]
+        flow_watermodels[t] = wm_solution["solution"]["nw"][string(t)][link_type][link_id]["q"]
+        start_node = wntr_data.links._data[link_name].start_node_name
+        end_node = wntr_data.links._data[link_name].end_node_name
+        head_loss_wntr[t] = wntr_simulation.node["head"][start_node].values[t]-wntr_simulation.node["head"][end_node].values[t]
+        head_loss_watermodels[t] = -1*wm_solution["solution"]["nw"][string(t)][link_type][link_id]["dhn"]+wm_solution["solution"]["nw"][string(t)][link_type][link_id]["dhp"]
+    end
 
+    link_df = DataFrame(time = 1:time_step:time_step*duration, flow_wntr = vec(flow_wntr), flow_watermodels = vec(flow_watermodels), 
+        head_loss_wntr = vec(head_loss_wntr), head_loss_watermodels = vec(head_loss_watermodels) )
+
+    return link_df
+end
+
+# compare pipes
+function get_pipe_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation,pipe_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    pipe_name = wm_data["nw"]["1"]["pipe"][pipe_id]["source_id"][2]
+    flow_wntr = Array{Float64,1}(undef,duration)
+    flow_watermodels = Array{Float64,1}(undef,duration)
+    head_loss_wntr = Array{Float64,1}(undef,duration)
+    head_loss_watermodels = Array{Float64,1}(undef,duration)
+
+    for t in 1:duration
+        flow_wntr[t] = wntr_simulation.link["flowrate"][pipe_name].values[t]
+        flow_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["pipe"][pipe_id]["q"]
+        start_node = wntr_data.links._data[pipe_name].start_node_name
+        end_node = wntr_data.links._data[pipe_name].end_node_name
+        head_loss_wntr[t] = wntr_simulation.node["head"][start_node].values[t]-wntr_simulation.node["head"][end_node].values[t]
+        head_loss_watermodels[t] = -1*wm_solution["solution"]["nw"][string(t)]["pipe"][pipe_id]["dhn"]+wm_solution["solution"]["nw"][string(t)]["pipe"][pipe_id]["dhp"]
```

Comment:
Same comment as above when computing head differences from the WaterModels solution.